### PR TITLE
Sprint2- ABDULLAH ZAMAN

### PR DESCRIPTION
**Pull request by abdullah2021skipq**

### CODE REVIEW

File: AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        # We define the schedule, target and the rule for our lambda
+        
+        lambda_schedule = events_.Schedule.rate(cdk.Duration.minutes(1))
+        lambda_target = targets_.LambdaFunction(handler=hw_lambda)
+        rule = events_.Rule(self, "WebHealth_Invocation", description = "Periodic Lambda",      # rule: which targets will get our event
+                            enabled=True, schedule=lambda_schedule, targets=[lambda_target])
+        
+        dynamo_table = self.create_table("AbdullahTable")
+        dynamo_table.grant_read_write_data(db_lambda)
+        # db_lambda.add_environment('table_name',"AbdullahTable")   ##No need for this
```
Comment (talha2021skipq) [inline AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py:49]: you commented that you do not need line#49.  I so then how did you pass the table name to lambda's environment?

File: AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        except:
+            pass
+    
+    def create_lambda(self, newid, asset, handler, role):
+        return lambda_.Function(self, id = newid,
+                                runtime = lambda_.Runtime.PYTHON_3_6,
+                                handler = handler,
+                                code = lambda_.Code.asset(asset),
+                                role=role,
+    			                    )
```
Comment (talha2021skipq) [inline AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py:124]: Still proper commenting can make it better for next time.

File: AbdullahZaman/sprint1/PCRepoAZB/resources/constants.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+URL_TO_MONITOR = "www.skipq.org"
```
Comment (talha2021skipq) [inline AbdullahZaman/sprint1/PCRepoAZB/resources/constants.py:1]: Now you are not taking URL from a file in S3 bucket. So we should optimize our program in compliance with memory efficiency as well.

File: AbdullahZaman/sprint2/PCRepoAZB/pc_repo_azb/pipeline_stack.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        beta = InfraStage(self, "AbdullahZBeta", env={
+        'account': '315997497220',
+        'region': 'us-east-2'
+        })
+        
+        gemma = InfraStage(self, "AbdullahZGemma", env={
+        'account': '315997497220',
+        'region': 'us-east-2'
+        })
+
```
Comment (talha2021skipq) [inline AbdullahZaman/sprint2/PCRepoAZB/pc_repo_azb/pipeline_stack.py:38]: Beta and Gamma stages are similar, so in my opinion, we do not need one of them, unless it has some unique functionality(step).

File: AbdullahZaman/sprint2/PCRepoAZB/unit_tests/unit_test2.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def test_alarm_count():
+    
+    app=core.App()
+    PcRepoAzbStack1(app,'AlarmCount')
+    template=app.synth().get_stack_by_name('AlarmCount').template
+    fun=[resource for resource in template['Resources'].values() if resource['Type']=='AWS::CloudWatch::Alarm']
+
+    assert len(fun)==2
```
Comment (talha2021skipq) [inline AbdullahZaman/sprint2/PCRepoAZB/unit_tests/unit_test2.py:13]: assert len(fun)==3, this should be the test, as we have 3 alarms in total on the cloudwatch.

**Review by talha2021skipq — APPROVED** [review]
A great job, but still can be improved to make it more readable by adding proper commenting. Moreover, I have added some comments regarding code logic, and code modularity at some points, kindly consider those comments if they help in improving it. Thank you!

File: AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py (hunk 1 above)
Comment (abdullah2021skipq) [reply AbdullahZaman/sprint1/PCRepoAZB/pc_repo_azb/pc_repo_azb_stack.py:49]: Corrected in the last commit.

File: AbdullahZaman/sprint1/PCRepoAZB/resources/constants.py (hunk 3 above)
Comment (abdullah2021skipq) [reply AbdullahZaman/sprint1/PCRepoAZB/resources/constants.py:1]: I was doing it for a single url, I will use the s3 bucket when I'm using multiple urls.

File: AbdullahZaman/sprint2/PCRepoAZB/unit_tests/unit_test2.py (hunk 5 above)
Comment (abdullah2021skipq) [reply AbdullahZaman/sprint2/PCRepoAZB/unit_tests/unit_test2.py]: Thank You. Resolved in the last commit

File: AbdullahZaman/sprint2/PCRepoAZB/pc_repo_azb/pipeline_stack.py (hunk 4 above)
Comment (abdullah2021skipq) [reply AbdullahZaman/sprint2/PCRepoAZB/pc_repo_azb/pipeline_stack.py:38]: I had to use integration and unit tests separately for separate environments, keeping in view the production stage.

**Pull request merged**